ext/scripts/install_scripts/install_ppa.pl

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/perl -w
+
+=pod
+ 
+=head1 SYNOPSIS
+  Add a new PPA and clean up afterwards.
+
+  install_ppa.pl [OPTIONS]
+  Options:
+    --help                Brief help message
+    --dry-run             Doesn't execute the command, only prints it to STDOUT
+    --ppa                 PPA to add.
+
+=cut
+
+use strict;
+use File::Basename;
+use lib dirname (__FILE__);
+use package_mgmt_utils;
+use Getopt::Long;
+
+my $help = 0;
+my $dry_run = 0;
+my $ppa = '';
+
+
+GetOptions (
+            "help" => \$help,
+            "dry-run" => \$dry_run,
+            "ppa=s" => \$ppa
+          ) or package_mgmt_utils::print_usage_and_abort(__FILE__,"Error in command line arguments",2);
+package_mgmt_utils::print_usage_and_abort(__FILE__,"",0) if $help;
+
+if($ppa eq ''){
+    package_mgmt_utils::print_usage_and_abort(__FILE__,"Error in command line arguments: --ppa was not specified",1);
+}
+
+print("PPA is $ppa");
+my $cmd = "add-apt-repository -y '$ppa'";
+
+my $ppa_provider = "software-properties-common";
+package_mgmt_utils::execute("apt-get -y update",$dry_run);
+package_mgmt_utils::execute("apt-get install -y $ppa_provider",$dry_run);
+package_mgmt_utils::execute($cmd,$dry_run);
+package_mgmt_utils::execute("apt-get -y --purge autoremove $ppa_provider",$dry_run);
+package_mgmt_utils::execute("locale-gen en_US.UTF-8",$dry_run);
+package_mgmt_utils::execute("update-locale LC_ALL=en_US.UTF-8",$dry_run);
+package_mgmt_utils::execute("apt-get -y update",$dry_run);
+package_mgmt_utils::execute("apt-get check",$dry_run);
+package_mgmt_utils::execute("apt-get -y -f install",$dry_run);
```

Comment:
where does this come from and what is the intention?

Reply:
https://askubuntu.com/questions/708701/using-apt-get-to-remove-packages-and-dependencies

Reply:
```
-f, --fix-broken
       Fix; attempt to correct a system with broken dependencies in place. This
       option, when used with install/remove, can omit any packages to permit APT
       to deduce a likely solution. If packages are specified, these have to
       completely correct the problem. The option is sometimes necessary when
       running APT for the first time; APT itself does not allow broken package
       dependencies to exist on a system. It is possible that a system's dependency
       structure can be so corrupt as to require manual intervention (which usually
       means using dselect(1) or dpkg --remove to eliminate some of the offending
       packages). Use of this option together with -m may produce an error in some
       situations. Configuration Item: APT::Get::Fix-Broken.
```